analyses/npl/10_historical_event_timeline.md

```diff
@@ -0,0 +1,320 @@
+    df_station_stats['recall'] = df_station_stats['TP'].astype(int) / (df_station_stats['TP'].astype(int) + df_station_stats['FN'].astype(int))
+    return df_station_stats
+```
+
+```python
+df_station_stats = get_station_stats(df_station_dict, "event_danger", rp=2)
+df_station_stats
+```
+
+```python
+df_station_stats = get_station_stats(df_station_dict, "event_rp2", rp=2)
+df_station_stats
+```
+
+```python
+df_station_stats = get_station_stats(df_station_dict, "event_danger", rp=5, percentiles=[50])
+df_station_stats
+```
+
+```python
+df_station_stats = get_station_stats(df_station_dict, "event_rp5", rp=5, percentiles=[50])
+df_station_stats
+```
+
+### Get list of historical activations
+
+```python
+# Go through each date and list the activations
+
+event_bools = {
+    'readiness': False,
+    'action': False,
+    #'rp2': False
+    'danger': False
+}
+rp = 2
+
+for station in STATIONS:
+    for percentile in FORECAST_PERCENTILE_LIST:
+        print(station)
+        print(100-percentile)
+        df_station = df_station_dict[station]#.copy().dropna(subset=["water_level"])
+        df_events = pd.DataFrame(columns=['event', 'leadtimes'])
+        rp_val = df_return_period.loc[rp, station]
+        for date, row in df_station.iterrows():
+            for event_type in event_bools.keys():
+                cname = f'event_{event_type}'
+                if event_type in ['readiness', 'action']:
+                    cname += f"_rp{rp}_p{percentile}"
+                if row[cname]:
+                    if not event_bools[event_type]:
+                        s = 'flood' if event_type == 'danger' else event_type + ' trigger'
+                        print(date, s)
+                        # Find lead time
+                        if event_type in ['readiness', 'action']:
+                            forecast_leadtimes = []
+                            for leadtime in LEADTIMES_BY_TRIGGER[event_type]:
+                                if row[f'forecast_lt{leadtime}_p{percentile}'] > rp_val:
+                                    forecast_leadtimes += [leadtime]
+                            df_events.loc[date, 'leadtimes'] = forecast_leadtimes
+                            #print(forecast_leadtimes)
+                    event_bools[event_type] = True
+                else:
+                    event_bools[event_type] = False
+```
+
+### Plots
+
+```python
+# Try to plot the single readiness TP, and see which percentiles capture it
+rp = 2
+x = ds_glofas_forecast_summary['Chisapani_v3'].sel(leadtime=4, percentile=[20, 25, 30, 35, 40, 45, 50])
+rp_val = df_return_period.loc[rp, 'Chisapani']
+
+
+fig, ax = plt.subplots()
+x.plot.line(hue='percentile', alpha=0.5)
+ax.set_xlim(np.datetime64('2009-08-10'), np.datetime64('2009-08-20'))
+ax.axhline(rp_val, c='k')
+ax.set_ylim(4000, 8000)
+```
+
+```python
+# Plot timeline of events
```

Comment:
These figures look really nice! Only suggestion would be to add in a legend to indicate what each of the shaded sections correspond to, but you might also have done this manually in the slides.

Reply:
There is a legend way at the bottom